### ISSUE TITLE
Can't integrate my strategy with hyperopt

### ISSUE THREAD

**Issue by povstyany** [freqtrade/freqtrade#1285]
## Step 1: Have you search for this issue before posting it?

Yes

## Step 2: Describe your environment

It doesn't matter
  
## Step 3: Describe the problem:

I can't set up my strategy with using of hyperopt syntax. How can I setup complex conditions that work together? To be short I have several branches of logic in my strategy. I have succesfully created strategy according to test_strategy.py template and tested it. But i can't rewrite it into hyperopt syntax.

### Steps to reproduce:

Copy attached hyperopt.py and confg.json into your folder
Run python ./freqtrade/main.py -c config.json hyperopt -e 50 --timerange=20180901-20180930
 
Attachments: https://fex.net/685947586472
### Observed Results:

An error occured

### Relevant code exceptions or logs:

  ```
ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
  ```

**Comment by xmatthias** [comment]
Hi there,

it's a syntax error, really,
you cannot compare dataframes (or series) with "and" - but you'll need to use the logical boolean operator (just as you need to do in the strategy file)

#### Wrong
``` python
 conditions.append(
                dataframe['SMA_short'] < dataframe['SMA_long'] and 
                dataframe['BEAR_RSI'] < params['BearRsiHight-value']
            )
```

#### Right
``` python
 conditions.append(
                (dataframe['SMA_short'] < dataframe['SMA_long']) &
                (dataframe['BEAR_RSI'] < params['BearRsiHight-value'])
            )
```

once you exchange this - the error disappears.

**Comment by povstyany** [comment]
I've tried. Result is the same
![image](https://user-images.githubusercontent.com/13405326/47063282-1ed72e00-d1e3-11e8-911d-0afdeccbbd09.png)

**Comment by xmatthias** [comment]
you need to put each comparison in parenteses for the operation to work (check my code above very carefully - i did not just replace and with & ...)

**Comment by povstyany** [comment]
Thank you for fast answers.
Looks like started to work and now fails with not related to lambda error. I'll investigate.

Could you please also clarify if I have the opportunity to optimize the indicators timeperiod parameters? 
As I understood, I setup indicators in method populate_indicators() and can't backtest with different timeperiod parameters. Am I right?

Under optimization of timeperiod I understand, for example, finding the best candles count for calculating SMA:
`dataframe['SMA_long'] = ta.SMA(dataframe, timeperiod=860) //want to search best timeperiod in range from 800 to 900.
`

**Comment by xmatthias** [comment]
that's currently not implemented, sorry.

the only spaces currently implemented are "roi", "stoploss", "buy". 

implementing that should be relatively simple - following the sample for the buy code (happy to get a PR for that) - however it's not very high on my list.

**Comment by povstyany** [comment]
Hi, thank you for answer!

I have few more points to clarify.

You did not implement "sell" space because freqtrade is targeted to trading especially by take profit, did you? 

I mean, for example, Gekko framework backtests strategies according to "buy"/"sell" signals and i was a bit surprized to see that selling by "sell" signal is disabled in freqtrade's config. I made a conclusion that frequtrade's hyperopt works to find the strategy that buys on the grounds of indicators values and sells especially using "take profit strategy" . Is my conclusion correct? If yes, why did you choose exaclty the way with take profit? Are these strategies more effective that strategies that sell on the grounds of indicators values? 

P.S.: I managed to run hyperopt succesfully and already got some results. Thank you for your job, it's great!

**Comment by xmatthias** [comment]
no, it's not implemented because noone took the time to do it so far.

It should be fairly easy to do (copy/paste the buy part) - but unless someone has the time and motivation to do so and submits a PR for it.

I think writing a strategy without sell-signal is easier for beginners (in addition to that, using a sell-signal wasn't supported initially) - but it is now (however disabled by default).

FreqTrade tries to be simple to get startet with - and looking at only one side of a trade (buy) makes it slightly easier to start with. Once you write more advanced strategies, you'll need and want sell-signal - so it's also supported.


I wouldn't dare to determine the effectiveness of a strategy by that - if you have a very good buy signal (buying right before a high rise) and good sell signal (right after the huge rise ends) - then the take-profit approach is not very suitable - as it will probably sell too early.

However I've not succeeded in writing that strategy yet ... feel free to let me know if you do ...

Even on a fairly simple strategy (SMA cross EMA for example) you can make profits - but if you use the same as sell-signal it'll probably sell too late... this is where the "take profit" / ROI approach works better.

summing up - it all depends on the strategy - and your method to approach this.

Going to close this as the initial problem was solved. 
feel free to join our [Slack channel](https://join.slack.com/t/highfrequencybot/shared_invite/enQtMjQ5NTM0OTYzMzY3LWMxYzE3M2MxNDdjMGM3ZTYwNzFjMGIwZGRjNTc3ZGU3MGE3NzdmZGMwNmU3NDM5ZTNmM2Y3NjRiNzk4NmM4OGE) for further discussions - we also have a channel for sharing strategy ideas ...
